src/dune/dune_file.ml

```diff
@@ -1486,7 +1487,23 @@ module Executables = struct
            User_error.raise ~loc
              [ Pp.text "This field is reserved for Dune itself" ];
          fname)
-    and+ enabled_if = enabled_if ~since:(Some (2, 3)) in
+    and+ enabled_if =
+      let allowed_vars =
+        Some
+          (List.map
+             ~f:(function
+               | var -> (var, (2, 6)))
+             [ "architecture"
+             ; "system"
+             ; "model"
+             ; "os_type"
+             ; "ccomp_type"
+             ; "profile"
+             ; "ocaml_version"
```

Comment:
Hmm, the duplication between this list and the one above is a bit unfortunate. Can we share the common elements?